src/platforms/android/common/gradle.mdx

```diff
@@ -0,0 +1,242 @@
+    }
+}
+```
+
+Now when you build your app, the plugin will upload the ProGuard/R8 mappings, source bundle, and native symbols as you configured them to Sentry.
+
+### Variant Filtering
+
+You can specify which variant/flavor/build-type should be ignored by Sentry in your `app/build.gradle` file:
+
+```groovy
+sentry {
+    // List the build types that should be ignored (e.g. "release").
+    ignoredBuildTypes = ["release"]
+
+    // List the build flavors that should be ignored (e.g. "production").
+    ignoredFlavors = ["production"]
+
+    // List the build variant that should be ignored (e.g. "productionRelease").
+    ignoredVariants = ["productionRelease"]
+}
+```
+
+```kotlin
+sentry {
+    // List the build types that should be ignored (e.g. "release").
+    ignoredBuildTypes.set(setOf("release"))
+
+    // List the build flavors that should be ignored (e.g. "production").
+    ignoredFlavors.set(setOf("production"))
+
+    // List the build variant that should be ignored (e.g. "productionRelease").
+    ignoredVariants.set(setOf("productionRelease"))
+}
+```
+
+## ProGuard/R8
+
+The plugin will automatically generate appropriate ProGuard mapping files and upload them when you run `gradle assemble{BuildVariant}`. For example, `assembleRelease` — Release is the default, but the plugin works for others if you have enabled ProGuard/R8. The credentials for the upload step are loaded via environment variables or from a `sentry.properties` file in your project root.
+
+For more information, see the [full sentry-cli documentation](/product/cli/configuration/#configuration-values).
+
+At the very minimum you will need something similar to:
+
+```bash
+defaults.project=your-project
+defaults.org=your-org
+auth.token=YOUR_AUTH_TOKEN
+```
+<Note>
+
+You can find your authentication token [on the Sentry API page](https://sentry.io/api/). For more information about the available configuration options, see the [full sentry-cli documentation](/product/cli/configuration/#configuration-values).
+
+</Note>
+
+## Native Debug Symbols and Sources
+
+The plugin will automatically discover and upload Native debug symbols and sources when you run `gradle assemble{BuildVariant}`. The `{BuildVariant}` must be a non-debuggable build type, for example `Release`. The credentials for the upload step are loaded via environment variables or from a `sentry.properties` file in your project root.
+
+For more information, see the [full sentry-cli documentation](/product/cli/configuration/#configuration-values).
+
+At the very minimum you will need something similar to:
+
+```bash
+defaults.project=your-project
+defaults.org=your-org
+auth.token=YOUR_AUTH_TOKEN
+```
+<Note>
+
+You can find your authentication token [on the Sentry API page](https://sentry.io/api/). For more information about the available configuration options, see the [full sentry-cli documentation](/product/cli/configuration/#configuration-values).
+
+</Note>
+
+## Tracing Auto-instrumentation
```

Comment:
@imatwawana and also this one especially, the rest was basically taken as-is, but of course feel free to change that as well :)